Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -379,29 +379,47 @@ public override void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.Event
             Debug.Assert(eventData != null);
             Debug.Assert(!(eventData is MixedRealityPointerEventData), "HandleEvent called with a pointer event. All events raised by pointer should call HandlePointerEvent");
 
-            var baseInputEventData = ExecuteEvents.ValidateEventData<BaseInputEventData>(eventData);
-            DispatchEventToGlobalListeners(baseInputEventData, eventHandler);
+            var baseEventData = ExecuteEvents.ValidateEventData<BaseEventData>(eventData);
+            DispatchEventToGlobalListeners(baseEventData, eventHandler);
 
-            if (baseInputEventData.used)
+            if (baseEventData.used)
             {
                 // All global listeners get a chance to see the event,
                 // but if any of them marked it used,
                 // we stop the event from going any further.
                 return;
             }
 
-            Debug.Assert(baseInputEventData.InputSource.Pointers != null, $"InputSource {baseInputEventData.InputSource.SourceName} doesn't have any registered pointers! Input Sources without pointers should use the GazeProvider's pointer as a default fallback.");
-
             var modalEventHandled = false;
-            // Get the focused object for each pointer of the event source
-            for (int i = 0; i < baseInputEventData.InputSource.Pointers.Length && !baseInputEventData.used; i++)
+
+            var baseInputEventData = eventData as BaseInputEventData;
```

Comment:
I don't, but I would guess the difference is that BaseInputEventData can hold InputSources which have multiple pointers. The FocusProvider on the other hand only handles registered pointers and can not provide the InputSource for the BaseInputEventData, hence uses the FocusEventData.
Fun fact: IMixedRealityFocusProvider provides all its implementation methods by pointer, which the Toolkit does not use anywhere. The only implementer of it is the FocusProvider and it registeres pointers by InputSource.
If it wasn't for the interface, there wouldn't be the need for a FocusEventData at all.
Only other issue would be that FocusExit needs the OldFocusedObject somehow, but this could go into BaseEventData selectedObject, which is also not used at all at the moment (assigment wise #5139).